provider/datagen/src/transform/cldr/characters/mod.rs

```diff
@@ -152,10 +211,14 @@ fn unescape_exemplar_chars(char_block: &str) -> String {
         panic!();
     };
 
-    ch_lite.trim().to_string()
+    let result = ch_lite.trim().to_string();
+
+    result
 }
 
-fn insert_chars_from_string(set: &mut HashSet<Cow<str>>, input: &str) {
+/// Parse the input string, and insert the represented exemplar "characters" (each of
+/// which could either be an individual code point or a code point sequence) into the set.
+fn insert_chars_from_string(set: &mut HashSet<String>, input: &str) {
```

Comment:
I created a separate method to handle the backslash parsing because I realized that they were used to denote a character literal.  (It makes sense for `{` and `}` and `-`, etc. which are elements of the UnicodeSet syntax.)  Previously I was just using `String::replace("\\\\...", "")`, but that was not correct, of course.  Because the following character needs to be added to the parsed element set and more importantly the substring needs to be excised from the input String, I'm mutating the String, so I couldn't any longer take in and modify in-place a `&str`.  Since I had to allocate anyways in this initial preprocessing step, and I wanted to keep the input & output types the same for that step's helper method, I didn't see the point of `Cow<str>` for the rest of the code. :-(

But if there's a better Rusty way to have gone about this, I'd like to learn.

Reply:
I guess you can keep the `&str` if it contains no escape sequences, but that seems to happen fairly rarely. So it's fine.